cmake/Modules/ArrowExternalProject.cmake

```diff
@@ -14,11 +14,11 @@
 #  - PLASMA_STATIC_LIB
 #  - PLASMA_SHARED_LIB
 
-set(arrow_URL https://github.com/apache/arrow.git)
-# The PR for this commit is https://github.com/apache/arrow/pull/3154. We
+set(arrow_URL https://github.com/ray-project/arrow.git)
```

Comment:
Can you document why we're cloning a fork and maybe include a link to the discussion in the arrow PR?